torchao/experimental/kernels/mps/src/OperationUtils.h

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) Meta Platforms, Inc. and affiliates.
+// All rights reserved.
+//
+// This source code is licensed under the license found in the
+// LICENSE file in the root directory of this source tree.
+
+#pragma once
+
+#include <iostream>
+
+static void fail(const std::string& str) {
+  std::cerr << str << std::endl;
+  abort();
+}
+
+static void fail(const std::string& str1, const std::string& str2) {
+  std::cerr << str1 << str2 << std::endl;
+  abort();
+}
+
+inline void dispatch_sync_with_rethrow(
+    id<MTLCommandQueue> queue,
+    void (^block)()) {
+  (void)queue;
+  block();
+}
+
+inline id<MTLDevice> getMetalDevice() {
+  NSArray* devices = [MTLCopyAllDevices() autorelease];
+  if (devices.count == 0) {
+    fail("Metal is not supported");
+  }
+  return devices[0];
```

Comment:
Consider adding autorelease here. Also do you compile with ARC or without it?
```suggestion
  @autorelese {
    NSArray* devices = [MTLCopyAllDevices() autorelease];
    if (devices.count == 0) {
      fail("Metal is not supported");
    }
    return devices[0];
  }
```